src/test/java/software/amazon/cloudformation/resource/ResourceTypeSchemaTest.java

```diff
@@ -210,6 +212,18 @@ public void removeWriteOnlyProperties_hasWriteOnlyProperties_shouldRemove() {
         assertThat(resourceModel.has("propertyB")).isTrue();
     }
 
+    @Test
+    public void removeWriteOnlyProperties_hasWriteOnlyProperties_shouldDoNothing() {
+        JSONObject o = loadJSON(TEST_SCHEMA_WITH_EMPTY_WRITE_ONLY_PATH);
+        ResourceTypeSchema schema = ResourceTypeSchema.load(o);
+        JSONObject resourceModel = loadJSON(EMPTY_WRITEONLY_MODEL_PATH);
+
+        schema.removeWriteOnlyProperties(resourceModel);
```

Comment:
still confused by the testing, doesn't make sense to me to call `removeWriteOnlyProperties` with another `resourceModel`, especially since that one doesn't have [`writeOnlyProperties`](https://github.com/aws-cloudformation/cloudformation-resource-schema#resource-semantics)
```suggestion
        JSONObject o = loadJSON(TEST_SCHEMA_WITH_EMPTY_WRITE_ONLY_PATH);
        ResourceTypeSchema schema = ResourceTypeSchema.load(o);

        schema.removeWriteOnlyProperties(o);
```

Reply:
Built resourcemodel from same schema.